src/librustc/lint/builtin.rs

```diff
@@ -500,6 +501,9 @@ impl BuiltinLintDiagnostics {
                     Applicability::MachineApplicable
                 );
             }
+            BuiltinLintDiagnostics::UnknownCrateTypes(span, note, sugg) => {
+                db.span_suggestion(span, &note, sugg);
```

Comment:
As noted in the prior discussion, use `span_suggestion_with_applicability`. `MaybeIncorrect` is _always_ a safe bet to use, `MachineApplicable` is not consistently used throughout the codebase. It sometimes means 100% certainty, while others it means "95% of the cases this error fires, this suggestion is correct". I'm ok with the overly restrictive reading of the meaning.